docs/en/development/io_processing.md

```diff
@@ -3,7 +3,51 @@ sidebar_label: Read/Write/Delete Processing Flow
+
+Generally, when writing a small file, the file will be uploaded to the object storage when it is being closed, and the I/O size is the same as the file size. It could also be seen from the third highlighted section (creating 128 KiB small files) in the above metrics figure that,
+
+- The PUT size to the object storage is 128 KiB, calculated by `object.put / object.put_c`.
+- The number of metadata transactions is two times to the number of PUT in approximate, since every file requires one create and one write.
+
+It's worth to be mentioned that, when uploading objects which are smaller than a block, JuiceFS will try to write them simultaneously into the local cache (specified by `--cache-dir`, memory and disk are all available) to keep it possible to improve performance in the future. From the metrics figure above, we can see that when writing small files, the write bandwidth of the blockcache is the same to that of the object. It is meaningful that reading these small files are extremely fast since they are cached, as shown in the fourth highlighted section.
+
+JuiceFS has very low write latency, about several microseconds in general. It is because a write operation will return immediately once it writes to the memory buffer of the client. It will not actually upload to the object storage until the upload action is triggered by internal events or by external applications. The internal events like the exceeding of the slice size, the slice number or the buffering time will trigger the upload action automatically, and the external application events such as closing a file or invoking `fsync` could also trigger the upload action. Since the buffer capacity will only be released after the data stored in the buffer is persisted, we may encounter a write block if the object storage is not fast enough to handle large-scale concurrencies to keep the buffer being released timely. Specifically, the default buffer size is 300 MiB, which can be set by the mount option `--buffer-size`. The usage of the buffer in realtime is shown in the `usage.buf` field in the metrics figure. JuiceFS client will prepend a 10ms delay to every write operation to slow down the operation if the buffer usage exceeds the threshold. New writes will be suspended until the buffer is released, if the usage of the buffer is over twice the threshold. Therefore, if the write latency keeps increasing or the buffer usage has exceeded the threshold for a long while, a larger `--buffer-size` may be needed. In addition, increasing `--max-uploads`, which is the maximum number of concurrencies that defaults to 20, could also be helpful to improve the bandwidth for writing to the object storage, thus boosting the buffer releasing.
```

Comment:
```suggestion
JuiceFS has very low write latency, about several microseconds in general. It is because a write operation will return immediately once it is done writing to the memory buffer of the client. It will not actually upload to the object storage until the upload action is triggered by internal events or by external applications. The internal events like the exceeding of the slice size, the slice number or the buffering time will trigger the upload action automatically, and the external application events such as closing a file or invoking `fsync` could also trigger the upload action. Since the buffer capacity will only be released after the data stored in the buffer is persisted, we may encounter a write block if the object storage is not fast enough to handle large-scale concurrencies to keep the buffer being released timely. Specifically, the default buffer size is 300 MiB, which can be set by the mount option `--buffer-size`. The usage of the buffer in realtime is shown in the `usage.buf` field in the metrics figure. JuiceFS client will prepend a 10ms delay to every write operation to slow down the operation if the buffer usage exceeds the threshold. New writes will be suspended until the buffer is released, if the usage of the buffer is over twice the threshold. Therefore, if the write latency keeps increasing or the buffer usage has exceeded the threshold for a long while, a larger `--buffer-size` may be needed. In addition, increasing `--max-uploads`, which is the maximum number of concurrencies that defaults to 20, could also be helpful to improve the bandwidth for writing to the object storage, thus boosting the buffer releasing.
```